src/models/TotalBalances.js

```diff
@@ -20,41 +20,46 @@
 
 import { BigNumber } from 'bignumber.js';
 
-// types
+// Types
 import type { AccountRecord } from 'models/Account';
+import type { AssetCategoryRecord } from 'models/AssetCategory';
 import type { ChainRecord } from 'models/Chain';
 
 /**
  * Total balances are represented in top-down nesting:
  *  - level 1. Account
- *  - level 2. Category
+ *  - level 2. Asset Category
  *  - level 3. Chain
  *
  * This hierarchy represents the order in which they are most frequently used.
  */
 
-// Category balances are calculated using selectors in redux
-export type CategoryRecord<T> = {
-  wallet?: T,
-  deposits?: T,
-  investments?: T,
-  liquidityPools?: T,
-  rewards?: T,
-};
-
-export type TotalBalances = AccountRecord<?AccountTotalBalances>;
-export type AccountTotalBalances = CategoryRecord<ChainRecord<BigNumber>>;
-
-// Stored balances are stored directly in redux
-export type StoreCategoryRecord<T> = {
-  deposits?: T,
-  investments?: T,
-  liquidityPools?: T,
-  rewards?: T,
-};
-
-export type StoreTotalBalances = AccountRecord<?StoreAccountTotalBalances>;
-export type StoreAccountTotalBalances = StoreCategoryRecord<ChainRecord<BigNumber>>;
-
-// Wallet balances are calculated using selectors in redux
-export type WalletTotalBalances = AccountRecord<ChainRecord<BigNumber>>;
+/** Generic helper type for Account -> Category -> Chain nesting. */
+export type AccountCategoryChainRecord<T> = AccountRecord<AssetCategoryRecord<ChainRecord<T>>>;
```

Comment:
😱

Reply:
Yup, this is our actual data structure, only renamed it for clarity